src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -245,19 +252,22 @@ internal sealed class TupleVirtualElementFieldSymbol : TupleElementFieldSymbol
         private readonly bool _cannotUse; // With LanguageVersion 7, we will produce named elements that should not be used
 
         public TupleVirtualElementFieldSymbol(
-            TupleTypeSymbol container,
+            NamedTypeSymbol container,
             FieldSymbol underlyingField,
             string name,
             int tupleElementIndex,
-            Location location,
+            ImmutableArray<Location> locations,
             bool cannotUse,
             bool isImplicitlyDeclared,
-            TupleElementFieldSymbol correspondingDefaultFieldOpt)
-
-            : base(container, underlyingField, tupleElementIndex, location, isImplicitlyDeclared, correspondingDefaultFieldOpt)
+            TupleElementFieldSymbol? correspondingDefaultFieldOpt)
+            : base(container, underlyingField, tupleElementIndex, locations, isImplicitlyDeclared, correspondingDefaultFieldOpt)
         {
-            Debug.Assert(name != null);
-            Debug.Assert(name != underlyingField.Name || !container.UnderlyingNamedType.Equals(underlyingField.ContainingType, TypeCompareKind.IgnoreDynamicAndTupleNames),
+            // The underlying field for 'Hanna' in a long tuple is Item1. The corresponding field is Item8.
```

Comment:
>'Hanna' [](start = 40, length = 7)

What does that name mean? Should this be "the 8-th item"?

Reply:
yes, many tuple tests use names by convention (Alice, Bob, etc)

---
In reply to: [355696221](https://github.com/dotnet/roslyn/pull/39370#discussion_r355696221) [](ancestors = 355696221)